File: 2003/CS101SamplesAll/CS101SamplesVSTO/UsingVBAandVSTOTogether/UsingVBAandVSTOTogether/EmployeeSelectorUserControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;

namespace LegacyTimesheet
{
    public partial class EmployeeSelectorUserControl : UserControl
    {
        public EmployeeSelectorUserControl()
        {
            InitializeComponent();

            this.vEmployeeTableAdapter.Fill(this.adventureWorks_DataDataSet.vEmployee);
        }

        private void selectButton_Click(object sender, EventArgs e)
        {
            AdventureWorks_DataDataSet.vEmployeeRow employeeRow = 
                (AdventureWorks_DataDataSet.vEmployeeRow)
                ((DataRowView)this.vEmployeeBindingSource.Current).Row;

            Globals.Sheet1.emailAddressNamedRange.Value =
                employeeRow.EmailAddress.Split('@')[0];
            Globals.Sheet1.employeeNameNamedRange.Value =
                employeeRow.LastName + ", " + employeeRow.FirstName;
            Globals.Sheet1.employeeNumberNamedRange.Value =employeeRow.EmployeeID;
            Globals.Sheet1.employeePhoneNamedRange.Value = employeeRow.Phone;
            Globals.Sheet1.employeeTitleNamedRange.Value = employeeRow.JobTitle;
            Globals.Sheet1.locationNamedRange.Value =
                employeeRow.City + ", " + employeeRow.StateProvinceName;
        }
    }
}

File: 2003/VB101SamplesAll/VB101SamplesVSTO/UsingVBAandVSTOTogether/UsingVBAandVSTOTogether/EmployeeSelectorUserControl.vb
Public Class EmployeeSelectorUserControl

    Public Sub New()

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        Me.vEmployeeTableAdapter.Fill(Me.adventureWorks_DataDataSet.vEmployee)

    End Sub
    Private Sub selectButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles selectButton.Click
        Dim employeeRow As AdventureWorks_DataDataSet.vEmployeeRow = _
            CType( _
            CType(Me.vEmployeeBindingSource.Current, DataRowView).Row(), _
            AdventureWorks_DataDataSet.vEmployeeRow)

        Globals.Sheet1.emailAddressNamedRange.Value = _
            employeeRow.EmailAddress.Split("@")(0)
        Globals.Sheet1.employeeNameNamedRange.Value = _
            employeeRow.LastName & ", " & employeeRow.FirstName
        Globals.Sheet1.employeeNumberNamedRange.Value = employeeRow.EmployeeID
        Globals.Sheet1.employeePhoneNamedRange.Value = employeeRow.Phone
        Globals.Sheet1.employeeTitleNamedRange.Value = employeeRow.JobTitle
        Globals.Sheet1.locationNamedRange.Value = _
            employeeRow.City & ", " & employeeRow.StateProvinceName

    End Sub
End Class
